tests/e2e/default/client-side-transitions/create-user-for-contacts.wdio-spec.js

```diff
@@ -0,0 +1,402 @@
+      await utils.saveDoc(district);
+
+      const originalContactId = await loginAsOfflineUser();
+
+      await commonPage.goToPeople(originalContactId);
+      await submitReplaceUserForm();
+
+      // Logout triggered immediately
+      await (await loginPage.loginButton()).waitForDisplayed();
+
+      await loginPage.cookieLogin();
+      await commonPage.goToReports();
+      const reportId = await getLastSubmittedReportId();
+
+      // Replace user report created
+      const replaceUserReport = await utils.getDoc(reportId);
+      assertReplaceUserReport(replaceUserReport, originalContactId);
+      const { replacement_contact_id } = replaceUserReport.fields;
+
+      await sentinelUtils.waitForSentinel(originalContactId);
+      const { transitions } = await sentinelUtils.getInfoDoc(originalContactId);
+
+      // Transition successful
+      expect(transitions.create_user_for_contacts.ok).to.be.true;
+      await assertOriginalContactUpdated(originalContactId, replacement_contact_id);
+      const originalContact = await assertOriginalContactUpdated(originalContactId, replacement_contact_id);
+      const newContact = await assertNewContact(ORIGINAL_USER, originalContact, replacement_contact_id);
+      await assertOriginalUserDisabled(originalContactId);
+      // Primary contact not updated
+      expect((await utils.getDoc(DISTRICT._id)).contact._id).to.equal('not-the-original-contact');
+      // New user created
+      const newUserSettings = await utils.getUserSettings({ contactId: replacement_contact_id });
+      newUsers.push(newUserSettings.name);
+      assertNewUserSettings(newUserSettings, newContact, ORIGINAL_USER);
+      const loginLink = await getTextedLoginLink(newUserSettings);
+
+      // Open the texted link
+      await commonPage.logout();
+      await browser.url(loginLink);
+      await commonPage.waitForPageLoaded();
+      const [cookie] = await browser.getCookies('userCtx');
+      expect(cookie.value).to.contain(newUserSettings.name);
+    });
+
+    it('does not create a new user or re-parent reports when the transition is disabled', async () => {
+      const settings = Object.assign({}, SETTINGS, { transitions: { create_user_for_contacts: false } });
+      await utils.updateSettings(settings, 'sentinel');
+      const originalContactId = await loginAsOfflineUser();
+
+      await commonPage.goToPeople(originalContactId);
+      await submitReplaceUserForm();
+      // No logout triggered
+      await commonPage.goToReports();
+      const reportId = await getLastSubmittedReportId();
+      const basicReportId = await submitBasicForm();
+
+      // Replace user report created
+      const replaceUserReport = await utils.getDoc(reportId);
+      assertReplaceUserReport(replaceUserReport, originalContactId);
+      const { replacement_contact_id } = replaceUserReport.fields;
+      // New contact created
+      const newContact = await utils.getDoc(replacement_contact_id);
```

Comment:
I am not sure if you are talking about in the context of this test, or in the context of the actual create-user-for-contacts functionality.  

In the context of the test, we need to use an offline user for the test to be relevant (and to prove that the only reason the transition did not happen was because it was disabled). AFIK the default test user is an online user, so I think we need to create one...

In the context of the actual functionality of the transition, the actual creation of the new contact (if that is indeed what you meant by _"creating this user unnecessarily"_ ) is somewhat tangential to the actual create-user-for-contacts functionality.  The `create-user-for-contacts.transition` does not control how the contact is created, but only cares if the proper id is given in the `replacement_contact_id` field on the configured replace form...  In this particular case, the new contact is created by the `replace_user` form and I am not sure if it is possible to check the `transitions` configuration from within the form to see if the transition is disabled or not....

Reply:
I think that when I was reading this, I thought that the transition was adding the new contact, instead of the form itself.